test/simplify_algebra_test.cpp

```diff
@@ -4133,6 +4134,46 @@ TEST_CASE(dot_slice_a)
     EXPECT(m1.sort() == m2.sort());
 }
 
+TEST_CASE(my_optim)
```

Comment:
This test case will test that the `find_mul_add` will not swap the `mul` and `add`. A good name for this test would be something like `skip_mul_add_for_horizontal_add_fusion`. Add another similar test case that tests `find_slice_add_mul` will swap if it finds the mul before add.